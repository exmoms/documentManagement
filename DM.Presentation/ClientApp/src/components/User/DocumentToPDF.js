import React from "react";
import { Document, Page, StyleSheet, Image, View } from "@react-pdf/renderer";

const styles = StyleSheet.create({
  page: {
    flexDirection: "row",
    backgroundColor: "#E4E4E4",
  },
  image: {
    width: "100%",
    height: "100%",
    padding: 10,
    backgroundColor: "white",
  },
});

export default function DocumentToPDF(props) {
  return (
    <Document>
      {props.scans.map((element, index) => {
        return (
          <Page key={index} size="A4" style={styles.page}>
            <View style={styles.image}>
              <Image src={element} />
            </View>
          </Page>
        );
      })}
    </Document>
  );
}
