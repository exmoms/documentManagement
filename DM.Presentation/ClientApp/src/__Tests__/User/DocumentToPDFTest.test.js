import React from "react";
import ReactDom from "react-dom";
import DocumentToPDF from "../../components/User/DocumentToPDF";
import { cleanup } from "@testing-library/react";
import { mount } from "enzyme";
import { shallow } from "enzyme";
import { act } from "react-dom/test-utils";
import { Document, Page, StyleSheet, Image, View } from "@react-pdf/renderer";
afterEach(cleanup);
describe("DocumentToPDF", () => {
    afterEach(() => {
        jest.clearAllMocks();
      });
  test("renders without crashing", () => {
    const div = document.createElement("div");
    let scans = ["ImageSrc1" , "ImageSrc2"];
    ReactDom.render(
      <DocumentToPDF
      scans={scans}
      />,
      div
    );
  });

  it("matches the snapshot", () => {
    let scans = ["ImageSrc1" , "ImageSrc2"];
    const wrapper = mount(
        <DocumentToPDF
      scans={scans}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("Form tests of Document History", () => {
    let scans = ["ImageSrc1" , "ImageSrc2"];
    const wrapper = mount(
        <DocumentToPDF
      scans={scans}
      />
    );
    expect(wrapper.find(Page).length).toBe(2);
    expect(wrapper.find(Image).length).toBe(2);
    expect(wrapper.find(Image).at(0).props().src).toBe("ImageSrc1");
    expect(wrapper.find(Image).at(1).props().src).toBe("ImageSrc2");
  });
});