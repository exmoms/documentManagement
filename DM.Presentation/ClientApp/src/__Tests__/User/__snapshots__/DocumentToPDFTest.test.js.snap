// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentToPDF matches the snapshot 1`] = `ReactWrapper {}`;
